waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -865,12 +872,13 @@
                                        deref)
         has-reverse-proxy? (when reverse-proxy
                              (envoy-sidecar-check-fn scheduler service-id service-description context))
-        offset (if has-reverse-proxy? 1 0)
         ;; Make $PORT0 value pseudo-random to ensure clients can't hardcode it.
         ;; Helps maintain compatibility with Marathon, where port assignment is dynamic.
-        service-port (-> service-id hash (mod 100) (* 10) (+ pod-base-port))
-        port0 (+ service-port offset)
-        health-check-port (+ port0 health-check-port-index)
+        service-id-hash (hash service-id)
+        service-port (service-id-hash->port0 service-id-hash pod-base-port)
+        port0 (if has-reverse-proxy?
+                (service-id-hash->port0 (inc service-id-hash) pod-base-port)
```

Comment:
Instead of `inc` can we pass an additional argument to `service-id-hash->port0`?
The incremented value is not a service hash.

Alternatively, we limit to 10 ports, so we can add a reasonable larger number after keeping sufficient buffer. E.g. `50` or `100`.

Reply:
I'm also renaming the function to `get-port-range` since `service-id-hash->port0` isn't very accurate.